### PR TITLE
NIFI-8727: Addressed bug in which ProcessSession doesn't properly dec…

### PR DESCRIPTION
**Pull request by markap14**
…rement claimant count when a FlowFile is cloned and then the clone written to. Added automated tests to ensure that we are properly handling cases where a FlowFile is clone and then the contents modified

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by markobean** [comment]
Applied this patch to a production flow which previously saw rapid and unbounded content repo growth. It was applied to two nodes in a cluster, one with RocksDBFlowFileRepository and one with WriteAheadFlowFileRepository. Both had content archiving disabled. The patched nodes showed a relatively flat content repo disk utilization. All other nodes showed a consistent, unbounded growth.
+1

**Comment by markobean** [comment]
The consequences of this issue are very significant. Inevitably, the disk housing the content repo will reach its capacity due to lack of content cleanup of flowfiles that have left the system; the disk space being used is not actually needed. This has been an issue since NiFi 1.12.1. Should we backport this patch to previous releases?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/repository/StandardProcessSessionIT.java [hunk 1]
```diff
@@ -2603,6 +2603,205 @@ public void testFullRollbackAFterCheckpointDoesNotStoreState() throws IOExceptio
...
+
+            final ContentClaim updatedCloneClaim = getContentClaim(clone);
+            assertEquals(1, contentRepo.getClaimantCount(updatedCloneClaim));
+            assertEquals(1, contentRepo.getClaimantCount(claim));
+        }
+    }
+
+    @Test
+    public void testCreateNewFlowFileWithoutParentThenMultipleWritesCountsClaimReferencesProperly() {
+        FlowFile clone = session.create();
```
Comment (theBestAndrew) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/repository/StandardProcessSessionIT.java:2790]: My understanding is that this is a new FlowFile, not a clone of a previous one, so should the variable name be changed?

**Comment by markap14** [comment]
@markobean thanks for testing & confirming. While you are obviously more than welcome to create your own personal build for your own use, within the apache community we generally do not backport patches to older releases. We simply include the fix in the next release, and users will need to ensure that they keep up-to-date in order to ensure that they have the latest & greatest.

**Comment by joewitt** [comment]
@markap14 changes look good and happy to merge but looks like the system tests tripped up - appears to be spurious so re-running them now.  Are those tests just unreliable on this strained infra?

**Comment by joewitt** [comment]
the system test issue appears related to them being flakey and that needs to be worked but not in this PR.

The PR appears to fix the issue noted.  +1 Merging.

**Pull request closed without merging**